torch/export/unflatten.py

```diff
@@ -1423,78 +1430,51 @@ class _IVals:
         """
-        Update attribute corresponding to a given intermediate value.
+        Read state corresponding to a given intermediate value.
         """
-        self.fqns[node.name].add(fqn)
-
-        # to update ival x, get attribute __ival__x and copy x to __ival__x
-        with graph.inserting_after(node):
-            ival_node = graph.get_attr("__ival__" + node.name, type_expr=node.type)
-            ival_node.meta = copy.copy(node.meta)
-        with graph.inserting_after(ival_node):
-            new_ival_node = graph.create_node(
-                "call_function", torch.ops.aten.copy_, (ival_node, node)
-            )
-            new_ival_node.meta = copy.copy(node.meta)
-
-    def create(self, partitions, root_module):
+        # we can assume that the node must be from a mutation
+        assert node.op == "call_function"
+        if not self._is_mutable(node.target):
+            # so the mutation was functionalized;
+            # we will apply the original mutation later (see below)
+            fqn, _ = next(reversed(node.meta["nn_module_stack"].values()))
+            self.node_names_by_fqn[fqn].add(node.name)
+        return mf.remap_input(node.args[0])
+
+    def update(self, partitions):
         """
-        Update attributes corresponding to intermediate values that were read.
-        Finally, initialize attributes in all modules that read or update
-        corresponding intermediate values.
+        Update states corresponding to intermediate values that were read.
         """
-
-        entries = [("", root_module)]
         for shared_submodules in partitions:
             for entry in shared_submodules:
-                entries.append((entry.fqn, entry.module))
                 graph = entry.module.graph
-                for node in graph.nodes:
-                    if node.name in self.storage:
-                        self.update(entry.fqn, graph, node)
-
-        # fqn -> list of ival node names read or updated through it
-        ivals = defaultdict(list)
-        for name, fqns in self.fqns.items():
-            for fqn in fqns:
-                ivals[fqn].append(name)
-
-        for fqn, mod in entries:
-            for name in ivals[fqn]:
-                ival_name = f"__ival__{name}"
-                # for a ival named x created in module call m,
-                # create attribute m.__ival__x, initially empty
-                setattr(mod, ival_name, self.storage[name])
+                node_names = self.node_names_by_fqn[entry.fqn]
+                nodes = [n for n in graph.nodes if n.name in node_names]
+                for node in nodes:
+                    # so node must be from a functionalized mutation;
+                    # we perform the original mutation now
+                    with graph.inserting_after(node):
+                        new_node = graph.create_node(
+                            "call_function",
+                            torch.ops.aten.copy_,
```

Comment:
should be ops.aten.copy_.default

Reply:
what's the difference? we use `aten.copy_` by itself elsewhere in this file

Reply:
aten.copy_ is not OpOverload i think. I guess it will do some schema mathcing thing in the runtime to find the correct overload?

Reply:
Gotcha